src/main/scala/chisel3/util/SparseVec.scala

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util
+
+import chisel3._
+import chisel3.experimental.SourceInfo
+import chisel3.util.{log2Up, BitPat, Mux1H}
+import chisel3.util.experimental.decode.{decoder, TruthTable}
+import scala.collection.mutable.{LinkedHashMap, LinkedHashSet}
+
+sealed trait SparseVec[A] {
+
+  def apply(addr: UInt)(implicit _sourceInfo: SourceInfo): A
+
+}
+
+object SparseVec {
+
+  def binary[A <: Data](
+    size:    Int,
```

Comment:
Is possible for size being inferred from mapping.size?

Reply:
Maybe. However, you may want to have to describe a `SparseVec` which is larger than what you have specified in the mapping. E.g., you only set element `1`, but you want to have a `SparseVec` that you can index with a `UInt(2.W)`. Maybe the address could be a `gen` parameter.

Reply:
There's an alternative constructor that now supports this.